code/src/containers/language-select/LanguageSelect.test.tsx

```diff
@@ -1,17 +1,23 @@
-import { act, fireEvent, screen } from "@testing-library/react";
+import { act, fireEvent, screen, waitFor } from "@testing-library/react";
 
 import LanguageSelect from "@/containers/language-select/LanguageSelect";
 
+import { LOCAL_STORAGE_KEYS } from "@/constants/common";
 import locales from "@/constants/locales";
 import renderWithProviders from "@/utils/render-with-providers/renderWithProviders";
 
+jest.spyOn(window.localStorage.__proto__, "setItem");
```

Comment:
`jest.spyOn(window.localStorage, "setItem")` does not work? As alternative we can use `spyOn(Storage.prototype, 'setItem')`, it look shorter:

https://stackoverflow.com/questions/11485420/how-to-mock-localstorage-in-javascript-unit-tests

Reply:
Thank you. For some reason, `jest.spyOn(window.localStorage, "setItem")` doesn't work. I changed that line with suggested `spyOn(Storage.prototype, 'setItem')`.